### ISSUE TITLE
GNOME Community translations

### ISSUE THREAD

**Issue by Sjoerd1993** [Sjoerd1993/Graphs#736]
Now we're part of Circle, [we are eligible](https://gitlab.gnome.org/Teams/Circle/-/blob/main/membership_guide.md) to enroll into the [GNOME Community Translations](https://l10n.gnome.org).

This by itself would probably make localization and such much easier, and would probably help a lot with getting people involved that are willing to contribute translations in different languages. However, there's a few caveats:

- One of the requirements is that we do not touch `.po` files ourselves, but instead let everything be handled by GNOME l10n. Tooling will compile changes to the `po` files automatically. This also means that we'd move away from Weblate, towards GNOME's official community translations. This is not really a big issue for me at least by the way.
- Releases need to be announced beforehand on the related discourse page, so that translators have time to update strings. Preferably two weeks before release, where strings should be frozen (broken strings can still be fixed). Again not an issue, and is only beneficial as it sets up a pretty natural period where there's strictly just testing.
- As far as I understand things, it seems like projects in this system need to be hosted on GitLab. We ware eligible to host Graphs on https://gitlab.gnome.org/World/. Personally I don't mind GitLab at all, but it would be a pretty disruptive move for quite obvious reasons. It would probably mean setting up the CI workflow all again, and I am not 100% sure if it would hurt discoverability of the repository. Although, either way I'd probably keep this repo up anyway as a mirror in such a case, but new MR's should then be handled there.

I am not 100% sure either way. The current situation works quite well, and it's a relatively disruptive move. But just wanted to throw the idea out there. Will probably try and inform myself a bit on the advantages either way on the long-term for the project.

Given the scale of such a move, I don't think it's a thing we should decide too hastily, and again it works quite well as-is. There's many other GNOME-related projects hosted on GitHub with Weblate for translations anyway.

**Comment by Sjoerd1993** [comment]
Yes, it's exactly those concerns why I host everything on GitHub myself, it's the de-facto standard and therefore offers the best discoverability. Also from a personal perspective, it's much simpler to link to my Github on a resume for instance, rather than to have it all on a more niche platform. But again, that's not really a factor for me when it comes to hosting Graphs. But it was a factor in the initial choice.

We're also accumulating some stars over time, and there exists links at some places towards the repo. Furthermore, it'd be a bit of a mess with issues and PR's. So again, it's quite disruptive and not something I'd do on a whim.

Purely from a FOSS perspective, GitHub being owned by Microsoft is not my favourite thing in the world, but so far it's been a good host for FOSS projects also after this acquisition by MS, so that has not been a deal-breaker for me.  Anyway, the main benefit would be that it'd put Graphs directly in the GNOME Translation workflow, and would thus be directly be part of the projects available to people who have signed up for the GNOME Translation teams. See the page with Circle apps here: https://l10n.gnome.org/languages/nl/gnome-circle/ui/ (For Dutch language). 

But again, in general Weblate is a good solution that works well, and moving repository is a disruptive move that should not really be done lightly. Will also look more into this. Perhaps it's possible to mirror Graphs to GitLab, translate from there, and push those changes into GitHub here. But I am not sure if such a hack/workaround is really viable or if that'd be too yanky, nor do I know if that'd be accepted by GNOME as a whole. So would need to dig more into that. Just wanted to throw the overall option out there for debate.

**Comment by Sjoerd1993** [comment]
So I asked, and it's a misconception that the Damned Lies translation system requires the repository to be hosted on GitLab. My impression that this was the case came from the default MR that is to be set up which has the GitLab project page as one of the fields:

![image](https://github.com/Sjoerd1993/Graphs/assets/68477016/a625cecc-a46e-421f-9ea6-37c7080052c3)

Either way, it works fine with GitHub. So if it's all-right, I'll probably set up a MR soon to get Graphs into the GNOME Translation system, as it does take away my biggest refrain from migrating from Weblate to GNOME's system. 

Regarding the repository host, I don't think a migration gives us any practical advantages, so I don't see any immediate reasons to consider moving that at the moment.

**Comment by cmkohnen** [comment]
On the other hand, since we've gotten this big release out there. Now might be the time to think about moving without much disruption.

moreover from what I've gathered in [this article](https://docs.gitlab.com/ee/user/project/import/github.html), it seems to be very trivial to import the repository including pull requests and issues and even migrate CI (as it seems enabled on GNOME's instance).
